closed/src/java.base/share/classes/com/sun/crypto/provider/NativeChaCha20Cipher.java

```diff
@@ -174,9 +174,13 @@ protected int engineGetOutputSize(int inputLen) {
         if (mode == MODE_NONE) {
             outLen = inputLen;
         } else if (mode == MODE_AEAD) {
-            outLen = (direction == Cipher.ENCRYPT_MODE) ?
-                    Math.addExact(inputLen, TAG_LENGTH) :
-                    Integer.max(inputLen - TAG_LENGTH, 0);
+            if (direction == Cipher.ENCRYPT_MODE) {
+                outLen = Math.addExact(inputLen, TAG_LENGTH);
+            }
+            else {
+                outLen = Integer.max(inputLen - TAG_LENGTH, 0);
```

Comment:
Perhaps this should be written as
```
outLen = Integer.max(inputLen, TAG_LENGTH) - TAG_LENGTH;
```
to avoid problems with large negative values for `inputLen`.